### DISCUSSION TITLE
NonEmpty + Swift-Collections (OrderedSet, Deque...)

### DISCUSSION THREAD

**Discussion by iDmitriyy** [pointfreeco/swift-nonempty#59] [Show and tell]
Hi

What about adding support of [swift-collections](https://github.com/apple/swift-collections) to NonEmpty? 
I suppose doing it with an overlay SPM module 'NonEmpty+UpcomingCollections'.

Currently, it is not possible to express `NonEmpty<OrderedDictionary>`, because OrderedDictionary doesn't conform to Collection protocol.
Another one problem with different Types of Dictionaries is impossibility to write generic Dictionary code. As a workaround, I've made NonEmptyOrderedDictionary implementation and added DictionaryUnifyingProtocol making possible common Dictionary operations in generic code with:  Swift.Dictionary, OrderedDictionary, NonEmptyDictionary, NonEmptyOrderedDictionary. There are also TreeDictionary, SortedDictionary. 
DictionaryUnifyingProtocol is something like _DictionaryProtocol in NonEmpty library with some differences.

With Set, OrderedSet, SortedSet, SparseSet + NonEmpty the approach is similar. For example, OrderedSet doesn't conform to SetAlgebraProtocol.

I can share code or make WIP pull request, but not sure how it is better to do.

**Comment by magnusburton** [comment]
Hi, please share the code that got this working for you.

Best,

**Comment by iDmitriyy** [comment]
@mbrandonw can youn please tell is such merge request will be accepted? In another discussion you've told that new features are not added to the library.  If a module with integration of NonEmpty + Swift-Collections can't be landed to this package, then I'll create a separated package, which is unfortunate though.

**Reply by mbrandonw** [reply]
Hi @iDmitriyy, we will not be accepting any PR's on this repo for now.
